pkg/instancetype/controller/vm/controller.go

```diff
@@ -116,6 +117,11 @@ func (c *controller) Sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineI
 		return vm, nil
 	}
 
+	// Before we sync ensure any referenced resources exist
+	if err := c.checkResourcesExist(vm); err != nil {
+		return vm, common.NewSyncError(err, common.FailedCreateVirtualMachineReason)
```

Comment:
Maybe use a different reason here? Did it actually fail to create a VirtualMachine?

Reply:
Failed to launch, I'll use something more related to this.

Reply:
I had missed the `syncError` to `Condition` translation the controller is already doing tbh, I'll rework this now and drop the custom condition handling.